tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -462,6 +462,74 @@ private enum CredentialType
             return app;
         }
 
+        [TestMethod]
+        public async Task CertificateOverrideAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                var cert = new X509Certificate2(ResourceHelper.GetTestResourceRelativePath("valid.crtfile"));
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                              .WithAuthority("https://login.microsoftonline.com/tid/")
+                              .WithExperimentalFeatures(true)
+                              .WithHttpManager(httpManager)
+                              .WithCertificate(cert)
+                              .Build();
+
+                Func<string, IReadOnlyList<KeyValuePair<string, string>>> clientAssertionOverride =
```

Comment:
I don't understand what the scenario is?

Reply:
The problem is it applies to the client assertions generated from the certificates. 
Can it (It should not) apply to the signed assertions passed instead of the client secret / cert?

Reply:
Ack, I'll move it to the app. 
The sceanrio here is to add an extensibility mechanism for SAL to do MsAuth_10_AT_POP, which I call "legacy POP"